### PR TITLE
update integration-cli/docker_cli_attach_unix_test.go to use Assert statements

### PR DESCRIPTION
**Pull request by ZJaffee**
As part of issue 16881

### CODE REVIEW

File: integration-cli/docker_cli_attach_unix_test.go [hunk 1]
```diff
@@ -19,9 +20,8 @@ func (s *DockerSuite) TestAttachClosedOnContainerStop(c *check.C) {
 	out, _ := dockerCmd(c, "run", "-dti", "busybox", "sleep", "2")
 
 	id := strings.TrimSpace(out)
-	if err := waitRun(id); err != nil {
-		c.Fatal(err)
-	}
+	err := waitRun(id)
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(waitRun(id), checker.IsNil)`

File: integration-cli/docker_cli_attach_unix_test.go [hunk 2]
```diff
@@ -74,9 +73,9 @@ func (s *DockerSuite) TestAttachAfterDetach(c *check.C) {
 	}()
 
 	time.Sleep(500 * time.Millisecond)
-	if err := waitRun(name); err != nil {
-		c.Fatal(err)
-	}
+	err := waitRun(name)
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(waitRun(id), checker.IsNil)`

File: integration-cli/docker_cli_attach_unix_test.go [hunk 3]
```diff
@@ -89,18 +88,15 @@ func (s *DockerSuite) TestAttachAfterDetach(c *check.C) {
...
 	cmd = exec.Command(dockerBinary, "attach", name)
 	cmd.Stdin = tty
 	cmd.Stdout = tty
 	cmd.Stderr = tty
 
-	if err := cmd.Start(); err != nil {
-		c.Fatal(err)
-	}
+	err := cmd.Start()
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(cmd.Start(), checker.IsNil)`

File: integration-cli/docker_cli_attach_unix_test.go [hunk 4]
```diff
@@ -123,9 +119,8 @@ func (s *DockerSuite) TestAttachAfterDetach(c *check.C) {
 		c.Fatal("timeout waiting for attach read")
 	}
 
-	if err := cmd.Wait(); err != nil {
-		c.Fatal(err)
-	}
+	err := cmd.Wait()
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(cmd.Wait(), checker.IsNil)`

File: integration-cli/docker_cli_attach_unix_test.go [hunk 5]
```diff
@@ -137,49 +132,40 @@ func (s *DockerSuite) TestAttachAfterDetach(c *check.C) {
 func (s *DockerSuite) TestAttachDetach(c *check.C) {
 	out, _ := dockerCmd(c, "run", "-itd", "busybox", "cat")
 	id := strings.TrimSpace(out)
-	if err := waitRun(id); err != nil {
-		c.Fatal(err)
-	}
+	err := waitRun(id)
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(waitRun(id), checker.IsNil)`

File: integration-cli/docker_cli_attach_unix_test.go [hunk 6]
```diff
@@ -137,49 +132,40 @@ func (s *DockerSuite) TestAttachAfterDetach(c *check.C) {
...
 	defer stdout.Close()
-	if err := cmd.Start(); err != nil {
-		c.Fatal(err)
-	}
-	if err := waitRun(id); err != nil {
-		c.Fatalf("error waiting for container to start: %v", err)
-	}
+	err := cmd.Start()
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(cmd.Start(), checker.IsNil)`

File: integration-cli/docker_cli_attach_unix_test.go [hunk 7]
```diff
@@ -137,49 +132,40 @@ func (s *DockerSuite) TestAttachAfterDetach(c *check.C) {
...
+
+	err := waitRun(id)
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(waitRun(id), checker.IsNil)`

File: integration-cli/docker_cli_attach_unix_test.go [hunk 8]
```diff
@@ -211,46 +196,39 @@ func (s *DockerSuite) TestAttachDetach(c *check.C) {
 func (s *DockerSuite) TestAttachDetachTruncatedID(c *check.C) {
 	out, _ := dockerCmd(c, "run", "-itd", "busybox", "cat")
 	id := stringid.TruncateID(strings.TrimSpace(out))
-	if err := waitRun(id); err != nil {
-		c.Fatal(err)
-	}
+	err := waitRun(id)
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(waitRun(id), checker.IsNil)`

File: integration-cli/docker_cli_attach_unix_test.go [hunk 9]
```diff
@@ -211,46 +196,39 @@ func (s *DockerSuite) TestAttachDetach(c *check.C) {
...
-		c.Fatal(err)
-	}
+	c.Assert(err, check.IsNil)
+	
 	defer stdout.Close()
-	if err := cmd.Start(); err != nil {
-		c.Fatal(err)
-	}
+	err := cmd.Start()
+	c.Assert(err, check.IsNil)
```
Comment (vdemeester) [inline integration-cli/docker_cli_attach_unix_test.go]: `c.Assert(cmd.Start(), checker.IsNil)`

**Comment by vdemeester** [comment]
@ZJaffee needs a rebase :sweat_smile:

**Comment by calavera** [comment]
I'm very sorry but I'm going to close this PR until the errors are resolved. Please, feel free to send it again after testing that every test still works and the code is valid. `make test` and `make validate` are your friends.

As a rule of thumb, before opening all your PRs again, it will be easier to get them merged if you focus in one at a time, that way reviewers don't need to point the same errors over several changes.

**Pull request closed without merging**